clang/test/SemaCXX/warn-thread-safety-parsing-system-macro.cpp

```diff
@@ -0,0 +1,65 @@
+// RUN: %clang_cc1 -fsyntax-only -I%S/Inputs -Wthread-safety %s 2>&1 | FileCheck %s
+// RUN: %clang_cc1 -fsyntax-only -I%S/Inputs -Wthread-safety -std=c++98 %s 2>&1 | FileCheck %s
+// RUN: %clang_cc1 -fsyntax-only -I%S/Inputs -Wthread-safety -std=c++11 %s -D CPP11 2>&1 | FileCheck %s
```

Comment:
@rapidsna Looks like there might be some missing test coverage here

* Late parsing isn't enabled in this test but the PR appears to be touching that code path
* There's no C version of this test that enables late parsing but this PR appears to be touching that code path.